server/datastore/mysql/disk_encryption_test.go

```diff
@@ -33,3 +40,51 @@ func testCleanupDiskEncryptionKeysOnTeamChange(t *testing.T, ds *Datastore) {
 	// No-op test
 	assert.NoError(t, ds.CleanupDiskEncryptionKeysOnTeamChange(ctx, []uint{1, 2, 3}, nil))
 }
+
+func testDeleteLUKSData(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+
+	hostOne, err := ds.NewHost(ctx, &fleet.Host{
+		DetailUpdatedAt: time.Now(),
+		LabelUpdatedAt:  time.Now(),
+		PolicyUpdatedAt: time.Now(),
+		SeenTime:        time.Now(),
+		NodeKey:         ptr.String("1"),
+		UUID:            "1",
+		Hostname:        "foo.local",
+		PrimaryIP:       "192.168.1.1",
+		PrimaryMac:      "30-65-EC-6F-C4-58",
+	})
+	require.NoError(t, err)
+
+	hostTwo, err := ds.NewHost(ctx, &fleet.Host{
+		DetailUpdatedAt: time.Now(),
+		LabelUpdatedAt:  time.Now(),
+		PolicyUpdatedAt: time.Now(),
+		SeenTime:        time.Now(),
+		NodeKey:         ptr.String("2"),
+		UUID:            "2",
+		Hostname:        "foo.local-zzz",
+		PrimaryIP:       "192.168.1.2",
+		PrimaryMac:      "30-65-EC-6F-C4-59",
+	})
+	require.NoError(t, err)
+
+	// Add a LUKS user key
+	randomBits := base64.StdEncoding.EncodeToString([]byte(uuid.New().String()))
+	var keySlot uint = 1
+
+	err = ds.SaveLUKSData(ctx, hostOne, randomBits, randomBits, keySlot)
+	require.NoError(t, err)
+
+	// Try to delete a non-existent LUKS key
+	err = ds.DeleteLUKSData(ctx, hostTwo.ID, keySlot)
+	require.NoError(t, err)
+
+	// Try to delete the wrong key slot
+	err = ds.DeleteLUKSData(ctx, hostOne.ID, keySlot+1)
+	require.NoError(t, err)
+
+	err = ds.DeleteLUKSData(ctx, hostOne.ID, keySlot)
+	require.NoError(t, err)
```

Comment:
Let's add a final check that there's no row after the `DeleteLUKSData` for that host.

Reply:
Done in https://github.com/fleetdm/fleet/pull/29034/commits/0e518dc9687f633b777effca1a8bb382b9817f5f